services/horizon/internal/ingest/processors/effects_processor.go

```diff
@@ -959,7 +959,20 @@ func (e *effectsWrapper) addClaimableBalanceEntryCreatedEffects(source *xdr.Muxe
 		history.EffectClaimableBalanceCreated,
 		details,
 	)
-	for _, c := range cb.Claimants {
+	// EffectClaimableBalanceClaimantCreated can be generated by
+	// `create_claimable_balance` but also by `liquidity_pool_withdraw` which
+	// also removes LP.
+	// In case of `create_claimable_balance` we use `op.Claimants` to make
+	// effects backward compatible. The reason for this is that Stellar-Core
+	// changes all `rel_before` predicated to `abs_before` so effects generated
+	// by Horizon 2.8.3 and 2.9.0 are different.
```

Comment:
Could you rephrase this so that future readers can understand it better? In particular what is `abs_before` and `rel_before` and why are they different depending on where we obtain the claimants from?

I have also proposed some other changes in the comment. Among other things I don't think that the versions are important, since we are releasing with backwards compatibility (but they do add to the confusion).

```suggestion
	// EffectClaimableBalanceClaimantCreated can be generated by
	// `create_claimable_balance` operation but also by `liquidity_pool_withdraw` operation
	// causing a revocation.
	// In case of `create_claimable_balance` we use `op.Claimants` to make
	// effects backward compatible. The reason for this is that Stellar-Core
	// changes all `rel_before` predicated to `abs_before`.
```

Reply:
Actually, including the explanation from the PR description would be enough:

> This is happening because all rel_before predicates present in operation body are changed to abs_before by Stellar-Core when the tx is included in the ledger.